test/sqllogictest/mzcompose.py

```diff
@@ -81,12 +81,22 @@ def run_sqllogictest(
         args.replicas, run_config, args, junit_report_path
     )
 
-    try:
-        for command in commands:
+    exceptions = []
+
+    for command in commands:
+        try:
             container_name = "sqllogictest"
             c.run(container_name, *command)
-    finally:
-        ci_util.upload_junit_report(c.name, MZ_ROOT / junit_report_path)
+        except Exception as e:
+            # Don't fail here so that we can keep running the remaining
+            # commands. We want to run through all SLT files to catch all
+            # current errors before failing the test.
+            exceptions.append(e)
+
+    ci_util.upload_junit_report(c.name, MZ_ROOT / junit_report_path)
+
+    if exceptions:
+        raise exceptions[0]
```

Comment:
What will happen to the other exceptions? Are they already "covered" by the junit.xml?

Reply:
Otherwise, we should print them

Reply:
The exceptions here are later swallowed anyway when they are caught. The only thing that seems to happen is that the calls to sqllogictest fail and we print the output.